# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_blur.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_blur.glsl
--- a/gfx/wr/webrender/res/cs_blur.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_blur.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -6,13 +6,13 @@
 
 #include shared,prim_shared
 
-varying vec2 vUv;
-flat varying vec4 vUvRect;
-flat varying vec2 vOffsetScale;
+varying highp vec2 vUv;
+flat varying highp vec4 vUvRect;
+flat varying mediump vec2 vOffsetScale;
 // The number of pixels on each end that we apply the blur filter over.
 // Packed in to vector to work around bug 1630356.
-flat varying ivec2 vSupport;
-flat varying vec2 vGaussCoefficients;
+flat varying mediump ivec2 vSupport;
+flat varying mediump vec2 vGaussCoefficients;
 
 #ifdef WR_VERTEX_SHADER
 // Applies a separable gaussian blur in one direction, as specified